snippets/csharp/language-reference/operators/EqualityAndNonEqualityExamples.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+
+namespace operators
+{
+    public static class EqualityAndNonEqualityExamples
+    {
+        public static void Examples()
+        {
+            Console.WriteLine("Value types:");
+            ValueTypesEquality();
+
+            Console.WriteLine("Strings:");
+            StringEquality();
+
+            Console.WriteLine("Reference types:");
+            ReferenceTypesEquality.Main();
+
+            Console.WriteLine("Non equality:");
+            NonEquality();
+        }
+
+        private static void ValueTypesEquality()
+        {
+            // <SnippetValueTypesEquality>
+            int a = 1 + 2 + 3;
+            int b = 6;
+            Console.WriteLine(a == b);  // output: True
+
+            char c1 = 'a';
+            char c2 = 'A';
+            Console.WriteLine(c1 == c2);  // output: False
+            Console.WriteLine(c1 == char.ToLower(c2));  // output: True
+            // </SnippetValueTypesEquality>
+        }
+
+        private static void StringEquality()
+        {
+            // <SnippetStringEquality>
+            string s1 = "hello!";
+            string s2 = "HeLLo!";
+            Console.WriteLine(s1 == s2.ToLower());  // output: True
+
+            string s3 = "Hello!";
+            Console.WriteLine(s1 == s3);  // output: False
+            // </SnippetStringEquality>
+        }
+
+        // Rationale for the next snippet.
+        // A method cannot contain a class definition. Thus, a standard way to include snippet doesn't work.
+        // We want snippets to be interactive. Thus, the whole snippet has a structure of the console program.
+        // (Running the code without the ReferenceTypesEquality class doesn't produce any output.)
+
+        // <SnippetReferenceTypesEquality>
+        public class ReferenceTypesEquality
+        {
+            public class MyClass
```

Comment:
The main idea of this snippet is to show that a reference type supports the `==` operator by default.